datadog/fwprovider/data_source_datadog_hosts.go

```diff
@@ -0,0 +1,273 @@
+		return
+	}
+
+	state.ID = types.StringValue("datadog-hosts")
+	state.TotalMatching = basetypes.NewInt64Value(ddHostListResponse.GetTotalMatching())
+	state.TotalReturned = basetypes.NewInt64Value(ddHostListResponse.GetTotalReturned())
+
+	var hostList []HostListModel
+	for _, val := range ddHostListResponse.HostList {
+		var hostListEntry HostListModel
+		hostListEntry.Aliases, _ = types.ListValueFrom(ctx, types.StringType, val.GetAliases())
+		hostListEntry.Apps, _ = types.ListValueFrom(ctx, types.StringType, val.GetApps())
+		hostListEntry.AWSName = basetypes.NewStringValue(val.GetAwsName())
+		hostListEntry.HostName = basetypes.NewStringValue(val.GetHostName())
+		hostListEntry.ID = basetypes.NewInt64Value(val.GetId())
+		hostListEntry.IsMuted = basetypes.NewBoolValue(val.GetIsMuted())
+		hostListEntry.LastReportedTime = basetypes.NewInt64Value(val.GetLastReportedTime())
+		hostListEntry.MuteTimeout = basetypes.NewInt64Value(val.GetMuteTimeout())
+		hostListEntry.Name = basetypes.NewStringValue(val.GetName())
+		hostListEntry.Sources, _ = types.ListValueFrom(ctx, types.StringType, val.GetSources())
+		hostListEntry.Up = basetypes.NewBoolValue(val.GetUp())
+		// TODO add metadata and metrics
+		if metrics, ok := val.GetMetricsOk(); ok {
+			hostListEntry.Metrics, _ = types.ObjectValueFrom(ctx, map[string]attr.Type{
+				"cpu":    types.Float64Type,
+				"iowait": types.Float64Type,
+				"load":   types.Float64Type,
+			}, HostListMetricsModel{
+				CPU:    basetypes.NewFloat64Value(metrics.GetCpu()),
+				IOWait: basetypes.NewFloat64Value(metrics.GetIowait()),
+				Load:   basetypes.NewFloat64Value(metrics.GetLoad()),
+			})
+		}
+		if meta, ok := val.GetMetaOk(); ok {
+			hostListEntry.Meta, _ = types.ObjectValueFrom(ctx, map[string]attr.Type{
+				"agent_version":   types.StringType,
+				"cpu_cores":       types.Int64Type,
+				"gohai":           types.StringType,
+				"machine":         types.StringType,
+				"platform":        types.StringType,
+				"processor":       types.StringType,
+				"python_version":  types.StringType,
+				"socket_fqdn":     types.StringType,
+				"socket_hostname": types.StringType,
+			}, HostListMetadataModel{
+				AgentVersion:    basetypes.NewStringValue(meta.GetAgentVersion()),
+				CPU_cores:       basetypes.NewInt64Value(meta.GetCpuCores()),
+				Gohai:           basetypes.NewStringValue(meta.GetGohai()),
+				Machine:         basetypes.NewStringValue(meta.GetMachine()),
+				Platform:        basetypes.NewStringValue(meta.GetPlatform()),
+				Processor:       basetypes.NewStringValue(meta.GetProcessor()),
+				PythonV:         basetypes.NewStringValue(meta.GetPythonV()),
+				Socket_FQDN:     basetypes.NewStringValue(meta.GetSocketFqdn()),
+				Socket_Hostname: basetypes.NewStringValue(meta.GetSocketHostname()),
+			})
+		}
+		hostList = append(hostList, hostListEntry)
+	}
+	state.HostList = hostList
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
```

Comment:
Did you mean to append errors when building the host entries? Since no errors are being appended, it seems like this check doesn't do much

Reply:
Oh duh. I don't think it makes sense to append errors for every single field in the host schema (a lot of them are sometimes missing depending on the specific host) so I'll just remove that error check and let those values be zero-value if the backend doesn't return it.